### ISSUE TITLE
Support for RANGE BETWEEN in window functions

### ISSUE THREAD

**Issue by carlshan** [prestodb/presto#15169]
**Background:**
Currently, Presto supports `ROWS BETWEEN` in doing window function aggregations. I.e. `SUM() OVER(ORDER BY date ROWS BETWEEN 6 PRECEDING AND CURRENT ROW)`.

This allows aggregations via direct reference to the row number. That is very useful.

However, Presto does not yet support aggregations via direct reference to the row value.

**Request:**
I often want to aggregate by a reference to the value of the row. This is what `RANGE BETWEEN` in other SQL dialects is supposed to do. For example, I may want to sum over the last 7 _days_, rather than the last 7 _rows_.

This feature is supported in other SQL implementations. In addition, Presto [documentation](https://prestodb.io/docs/current/functions/window.html) currently makes passing mention of `RANGE` but do not show any examples.

**Current implementation limitations:**

However, Presto `RANGE` seems to be limited to "UNBOUNDED" and "CURRENT ROW" with no other possible parameters.

According to this [Stack Overflow question](https://stackoverflow.com/questions/60302379/difference-between-rows-between-and-range-between-in-presto-window-functio) by someone who investigated the Presto code support for `RANGE`:

> The Presto code has a parser and test cases for the ROWS variant, but there's no mention in the documentation of ROWS.
>
> The test cases I found with both ROWS and RANGE don't test anything different between the two syntaxes.
> 
> They almost look like synonyms, but they do behave differently in my testing, and have different allowed parameters and validation rules.

Is this feature currently in any roadmap? If so, what might be the rough timeline for availability?

If not, I would like to suggest support for this feature as it could greatly help in performing many types of aggregations.

Thank you.

**Comment by taylorbentz** [comment]
I agree this would make my queries FAR more efficient, as I have to cross join all possible dates, multiplying the dataset by 20-30X it's actual size, because I cannot say "range between month(my_date) 1 and month(my_date) 12." This would be so much more efficient. I miss having that ability in BigQuery Standard SQL
